usr/sbin/regionsrv-enabler-azure

```diff
@@ -12,6 +12,35 @@
 # Lesser General Public License for more details.
 # You should have received a copy of the GNU Lesser General Public
 # License along with this library.
+
+"""
+Check the different states of the license type:
+    START   LICENSE CHANGED TO     CACHE   ACTION
+    BYOS-1  PAYG                   No      Register - Update Infra
+    BYOS-1  PAYG                   BYOS    Register - Update Infra
+    BYOS-1  BYOS-1                 No      Do nothing
+    BYOS-1  BYOS-1                 BYOS    Do nothing
+    BYOS-1  BYOS-2                 No      Do nothing
+    BYOS-1  BYOS-2                 BYOS    Do nothing
+
+    BYOS-2  PAYG                   No      Do nothing
+    BYOS-2  PAYG                   BYOS    Re-register - Update Infra
+    BYOS-2  BYOS-1                 No      De-register - Update Infra
+    BYOS-2  BYOS-1                 BYOS    De-register - Update Infra
+
+    PAYG    BYOS-1                 No      De-register - Update Infra
+    PAYG    BYOS-1                 No      De-register - Update Infra
+    PAYG    BYOS-2                 No      Do nothing
+    PAYG    BYOS-2                 PAYG    Do nothing
+    PAYG    PAYG                   No      Do nothing
+    PAYG    PAYG                   PAYG    Do nothing
+
+BYOS-1 -> BYOS unregistered
+BYOS-2 -> BYOS registered
```

Comment:
We need to consider the a `BYOS-3` case where `BYOS-3` is defined as registered to SCC and we need to handle that transition for example

`BYOS-3  PAYG  No  dergister from SCC register with update infrastructure.`

is one of the state transitions we need to handle.

It may also be a good idea to number the state transitions and then in the implementation reference the numbers the various if conditions handle.

Let's focus on getting the state transitions captured and then take another look at the implementation.